servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/AfterFinallyCompletable.java

```diff
@@ -23,43 +23,49 @@
 
 final class AfterFinallyCompletable extends AbstractSynchronousCompletableOperator {
 
-    private final Runnable runnable;
+    private final TerminalSignalConsumer doFinally;
 
-    AfterFinallyCompletable(Completable original, Runnable runnable, Executor executor) {
+    AfterFinallyCompletable(Completable original, TerminalSignalConsumer doFinally, Executor executor) {
         super(original, executor);
-        this.runnable = requireNonNull(runnable);
+        this.doFinally = requireNonNull(doFinally);
     }
 
     @Override
     public Subscriber apply(final Subscriber subscriber) {
-        return new AfterFinallyCompletableSubscriber(subscriber, runnable);
+        return new AfterFinallyCompletableSubscriber(subscriber, doFinally);
     }
 
     private static final class AfterFinallyCompletableSubscriber implements Subscriber {
         private final Subscriber original;
-        private final Runnable runnable;
+        private final TerminalSignalConsumer doFinally;
 
-        private static final AtomicIntegerFieldUpdater<AfterFinallyCompletableSubscriber> completeUpdater =
-                AtomicIntegerFieldUpdater.newUpdater(AfterFinallyCompletableSubscriber.class, "complete");
+        private static final AtomicIntegerFieldUpdater<AfterFinallyCompletableSubscriber> doneUpdater =
+                AtomicIntegerFieldUpdater.newUpdater(AfterFinallyCompletableSubscriber.class, "done");
         @SuppressWarnings("unused")
-        private volatile int complete;
+        private volatile int done;
 
-        AfterFinallyCompletableSubscriber(Subscriber original, Runnable runnable) {
+        AfterFinallyCompletableSubscriber(Subscriber original, TerminalSignalConsumer doFinally) {
             this.original = original;
-            this.runnable = runnable;
+            this.doFinally = doFinally;
         }
 
         @Override
         public void onSubscribe(Cancellable originalCancellable) {
-            original.onSubscribe(new BeforeCancellable(originalCancellable, this::afterFinally));
+            original.onSubscribe(new BeforeCancellable(originalCancellable, () -> {
```

Comment:
(pre-existing): `BeforeCancellable` -> consider renaming to `ComposedCancellable` or as the "before" naming convention clashes with the different operators that use is (e.g. this is an `After` operator that uses a `Before` class).